### PR TITLE
TBD: (need to resolve cycle issue ) - Feature/enhance twingate group

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/65

## Changes
- added new optional fields for group resource: users and resources

### CODE REVIEW

File: twingate/resource_group.go [hunk 1]
```diff
@@ -24,6 +25,18 @@ func resourceGroup() *schema.Resource {
 				Required:    true,
 				Description: "The name of the group",
 			},
+			"users": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "List of user IDs",
```
Comment (vmanilo) [inline twingate/resource_group.go:32]: @alexmensch please check description for new optional fields

File: twingate/resource_group.go [hunk 2]
```diff
@@ -24,6 +25,18 @@ func resourceGroup() *schema.Resource {
...
+			},
+			"resources": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "List of resource IDs",
```
Comment (vmanilo) [inline twingate/resource_group.go:38]: @alexmensch please check description for new optional fields

**Review by alexmensch — APPROVED** [review]
Documentation looks good to me -- I made a commit for the changes, please double-check that I did it correctly. :)

File: twingate/resource_connector_test.go [hunk 3]
```diff
@@ -68,7 +68,14 @@ func TestAccRemoteConnectorWithCustomName(t *testing.T) {
 					Config: createConnectorC2(remoteNetworkName, connectorName),
 					Check: ComposeTestCheckFunc(
 						testAccCheckTwingateConnectorExists(theResource, "twingate_remote_network.test_c2"),
-						resource.TestMatchResourceAttr(theResource, "name", regexp.MustCompile(connectorName)),
+						resource.TestCheckResourceAttr(theResource, "name", connectorName),
+					),
+				},
+				{
+					Config: createConnectorC2(remoteNetworkName, connectorName+"_"),
+					Check: resource.ComposeTestCheckFunc(
```
Comment (ekampf) [inline twingate/resource_connector_test.go:76]: Use our new compose func

**Comment by ekampf** [comment]
As discussed on our meeting we're going to come up with a different design for Milestone 3 that wont have these cyclic dependencies issue

**Pull request closed without merging**